core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -710,6 +708,17 @@ private[zio] final class FiberContext[E, A](
                     platform.logger(currentLocation, fiberId, logLevel, zio.message, contextMap, spans)
 
                     curZio = nextInstr(())
+
+                  case ZIO.Tags.OnPlatform =>
+                    val zio = curZio.asInstanceOf[ZIO.OnPlatform[Any, Any, Any]]
+
+                    val oldPlatform = platform
+
+                    platform = zio.platform()
+
+                    ensure(ZIO.succeed { platform = oldPlatform })
```

Comment:
If you want to be slower about this, but also simplify `FiberContext`, you could have `SetPlatform`, and do the ensuring bit with acquireRelease. It will be slower, for sure, albeit changing the platform is not something that we would expect to occur with any frequency.

Reply:
This would also be more compositional because it would open the door to implementing an operator to just set the platform which would then allow higher level operators to be defined for `ZManaged` and `ZStream` that allowed changing the platform for a scope.

Reply:
Exactly

Reply:
Done.